### PR TITLE
feat: Weight percentiles report #512

### PR DESCRIPTION
**Pull request by barakplasma**
- closes https://github.com/babybuddy/babybuddy/issues/512
- instead of adding the sex of the child to the DB model, or adding units to the weight report, I added new links to specific reports for boys and girls
- also bumps plotly versions
- adds quality of developer experience changes to the devcontainer / codespace extensions
- adds pre-commit black formatting / linting hook for devenv users

### CODE REVIEW

File: reports/graphs/weight_change.py
Comment (cdubz) [inline reports/graphs/weight_change.py:1]: Can we limit the traces to only go as far as their is weight data? Having the full set in there causes some oddness in the graph:


https://github.com/babybuddy/babybuddy/assets/10456740/c0406348-328a-460f-a4d3-fe3e5fd8686f

**Review by cdubz — CHANGES_REQUESTED** [review]
When I open the PR in codspaces the `.devcontainer.json` file has modifications after everything boots up --

```diff
@cdubz ➜ /workspaces/babybuddy (weight-percentiles-in-weight-graph ✗) $ git diff .devcontainer.json 
diff --git a/.devcontainer.json b/.devcontainer.json
index 75877d8..7110e18 100644
--- a/.devcontainer.json
+++ b/.devcontainer.json
@@ -4,7 +4,8 @@
       "extensions": [
         "bbenoist.Nix",
         "ms-python.python",
-        "batisteo.vscode-django"
+        "batisteo.vscode-django",
+        "ms-python.black-formatter"
       ]
     }
   },
```

File: reports/graphs/weight_change.py
Comment (barakplasma) [reply reports/graphs/weight_change.py:1]: of course

Comment (barakplasma) [reply reports/graphs/weight_change.py:1]: fixed in 1e8834c2078a4f5a79a8486213e3c098d0e279b8

**Comment by barakplasma** [comment]
> When I open the PR in codspaces the `.devcontainer.json` file has modifications after everything boots up --
> 
> ```diff
> @cdubz ➜ /workspaces/babybuddy (weight-percentiles-in-weight-graph ✗) $ git diff .devcontainer.json 
> diff --git a/.devcontainer.json b/.devcontainer.json
> index 75877d8..7110e18 100644
> --- a/.devcontainer.json
> +++ b/.devcontainer.json
> @@ -4,7 +4,8 @@
>        "extensions": [
>          "bbenoist.Nix",
>          "ms-python.python",
> -        "batisteo.vscode-django"
> +        "batisteo.vscode-django",
> +        "ms-python.black-formatter"
>        ]
>      }
>    },
> ```

fixed as well

**Review by cdubz — APPROVED** [review]

**Pull request merged**